x-pack/plugins/maps/public/classes/styles/vector/properties/static_icon_property.ts

```diff
@@ -8,13 +8,15 @@
 import type { Map as MbMap } from '@kbn/mapbox-gl';
 import { StaticStyleProperty } from './static_style_property';
 // @ts-expect-error
-import { getMakiSymbolAnchor, getMakiIconId } from '../symbol_utils';
+import { getMakiSymbolAnchor, getMakiIconId, loadMakiIconInMap } from '../symbol_utils';
 import { IconStaticOptions } from '../../../../../common/descriptor_types';
 
 export class StaticIconProperty extends StaticStyleProperty<IconStaticOptions> {
   syncIconWithMb(symbolLayerId: string, mbMap: MbMap, iconPixelSize: number) {
     const symbolId = this._options.value;
-    mbMap.setLayoutProperty(symbolLayerId, 'icon-anchor', getMakiSymbolAnchor(symbolId));
-    mbMap.setLayoutProperty(symbolLayerId, 'icon-image', getMakiIconId(symbolId, iconPixelSize));
+    loadMakiIconInMap(symbolId, mbMap).then(() => {
```

Comment:
Can you add logic to handle exceptions. For example, `loadMakiIconInMap` could throw an error if `getMakiSymbolSvg` gets an unknown id. I think the best way to handle the error is to surface a warning toast.